cmd/thanos/sidecar.go

```diff
@@ -214,6 +229,70 @@ func runSidecar(
 		mux := http.NewServeMux()
 		registerMetrics(mux, reg)
 		registerProfile(mux)
+		if uploads {
+			mux.HandleFunc("/snapshot", func(w http.ResponseWriter, r *http.Request) {
```

Comment:
maybe it is time to actually move this to separate func (:

Reply:
Agreed - will probably want to pass in a shared shipper as they will need to share some logic as well.